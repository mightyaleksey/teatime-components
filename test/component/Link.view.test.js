'use strict';

const Link = require('../../component/Link');
const React = require('react');
const renderer = require('react-test-renderer');

test('defaultProp #size should have a value `m`', () => {
  expect(Link.defaultProps.size).toBe('m');
});

test('Renders correctly with default props', () => {
  const tree = renderer.create(
    <Link>
      Link Content
    </Link>
  );

  expect(tree).toMatchSnapshot();
});

test('Renders correctly with defined props', () => {
  const tree = renderer.create(
    <Link
      className='outer'
      disabled
      download={true}
      href='#'
      id='links id'
      target='_blank'
      type='video/mp4'>
      Link Content
    </Link>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
