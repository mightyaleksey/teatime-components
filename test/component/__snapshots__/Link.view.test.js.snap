// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with default props 1`] = `
<a
  className="control"
  download={undefined}
  href={undefined}
  id={undefined}
  onClick={[Function]}
  target={undefined}
  type={undefined}
>
  Link Content
</a>
`;

exports[`Renders correctly with defined props 1`] = `
<a
  className="control disabled outer"
  download={true}
  href="#"
  id="links id"
  onClick={[Function]}
  target="_blank"
  type="video/mp4"
>
  Link Content
</a>
`;
